src/standalone/Soil/soil_hydrology_parameterizations.jl

```diff
@@ -7,7 +7,108 @@ export volumetric_liquid_fraction,
+end
+
+
+"""
+    update_albedo!(bc::AtmosDrivenFluxBC, p, soil_domain, model_parameters)
+
+Calculates and updates PAR and NIR albedo as a function of volumetric soil water content at
+the top of the soil. If the soil layers are very large, the water content of the top layer
+is used in the calclulation
+"""
+function update_albedo!(bc::AtmosDrivenFluxBC, p, soil_domain, model_parameters)
+    (;
+        ν,
+        θ_r,
+        PAR_albedo_dry,
+        NIR_albedo_dry,
+        PAR_albedo_wet,
+        NIR_albedo_wet,
+        alebdo_calc_top_thickness,
+    ) = model_parameters
+    # ∫H_S_e_dz is the integral of effective saturation from (surface-alebdo_calc_top_thickness) to surface
+    ∫H_S_e_dz = p.soil.sfc_θ_l
+    FT = eltype(soil_domain.fields.Δz_top)
+    # checks if there is at least 1 layer centered within the top soil depth
+    if maximum(soil_domain.fields.Δz_top) < FT(0.5) * alebdo_calc_top_thickness
+        # ∫H_dz is integral of 1 from (surface-alebdo_calc_top_thickness) to surface
+        ∫H_dz = p.soil.sfc_scratch
+        # calculate depths of layer centers
+        @. p.soil.sub_sfc_scratch =
+            soil_domain.fields.z_sfc - soil_domain.fields.z
+        # zero all centers lower than boundry, set everything above to one
+        @. p.soil.sub_sfc_scratch =
+            ClimaLand.heaviside.(
+                FT(0.5) * alebdo_calc_top_thickness + sqrt(eps(FT)),
+                p.soil.sub_sfc_scratch,
+            )
+        ClimaCore.Operators.column_integral_definite!(
+            ∫H_dz,
+            p.soil.sub_sfc_scratch,
+        )
+        # calculate depths again
+        @. p.soil.sub_sfc_scratch =
+            soil_domain.fields.z_sfc - soil_domain.fields.z
+        # zeros all θ_l lower than boundry
+        @. p.soil.sub_sfc_scratch =
+            ClimaLand.heaviside.(
+                FT(0.5) * alebdo_calc_top_thickness + sqrt(eps(FT)),
+                p.soil.sub_sfc_scratch,
+            ) * effective_saturation(ν, p.soil.θ_l, θ_r)
+        ClimaCore.Operators.column_integral_definite!(
+            ∫H_S_e_dz,
+            p.soil.sub_sfc_scratch,
+        )
+    else
+        # in the case where no layer is centered above boundry, use the top layer
+        ∫H_dz = soil_domain.fields.Δz_top
+        @. p.soil.sfc_scratch = soil_domain.fields.Δz_top + sqrt(eps(FT))
+        @. p.soil.sub_sfc_scratch =
+            soil_domain.fields.z_sfc - soil_domain.fields.z
+        # zeros all θ_l other than surface
+        @. p.soil.sub_sfc_scratch =
+            ClimaLand.heaviside.(p.soil.sfc_scratch, p.soil.sub_sfc_scratch) *
+            effective_saturation(ν, p.soil.θ_l, θ_r)
+        ClimaCore.Operators.column_integral_definite!(
+            ∫H_S_e_dz,
+            p.soil.sub_sfc_scratch,
+        )
+        # here ∫H_S_e_dz is equivalent to the effective_saturation of the top layer
+    end
+    @. p.soil.sfc_scratch = ∫H_S_e_dz / ∫H_dz
+    @. p.soil.PAR_albedo =
```

Comment:
It also might be more readable to say
```
S_sfc = p.soil.sfc_scratch
@. S_sfc = ∫H_S_e_dz / ∫H_dz
```
and use S_sfc in the below

Reply:
done